src/compiler/emitter.ts

```diff
@@ -3925,6 +3925,12 @@ module ts {
                     write(", ");
                     emit(imp.name);
                 });
+                forEach(node.amdDependencies, amdDependency => {

```

Comment:
Giving random names would add unnecessary garbage to emitted code. Since import order does not matter for AMD modules, reordering would make most sense.

Changing order is a bit intrusive to original code, but amd-dependency comment is not a language feature, just a utility. Its also not documented in official materials, so I think violating some project principles is justifiable.